client/src/main/java/edu/colostate/cs/cs414/chesshireCoders/jungleClient/app/Game/GameBoard.java

```diff
@@ -0,0 +1,275 @@
+		GamePiece gamePiece = gamePieces[color][piece];
+		int pieceColumn = gamePiece.getColumn();
+		int pieceRow = gamePiece.getRow();
+		
+		//Check the piece is not dead
+		if(gamePieces[color][piece] == null){
+			return false;
+		}
+		
+		//Only the row or column can change, one must change.
+		if(((column == pieceColumn) && (row == pieceRow))||((column != pieceColumn) && (row != pieceRow))){
+			return false;
+		}
+		
+		//The only time the row or column will increase by more than one is if the tiger or leopard is jumping the river
+		//Check if it increases by more than one
+		if(!(((pieceColumn+1 == column)||(pieceColumn-1 == column))||((pieceRow+1 == row)||(pieceRow-1 == row)))){
+			//if it is not a tiger or a leopard return false
+			if(!((piece == 4)||(piece == 5))) {
+				return false;
+			}
+			//if it is not jumping the left river it must jump right.
+			if(!(((pieceColumn == 0) && (column == 3)) || ((pieceColumn == 3) && (column == 0)) || ((pieceRow == 2) && (row == 6)) || ((pieceRow == 6) && (row == 2)))) {
+				//if it is not jumping the right river return false;
+				if(!(((pieceColumn == 3) && (column == 6)) || ((pieceColumn == 6) && (column == 3)) || ((pieceRow == 2) && (row == 6)) || ((pieceRow == 6) && (row == 2)))) {
+					return false;
+				}
+			}
+			
+			//If the tiger or leopard is jumping the river there can not be a rat in the way.
+			//We first need to check where the animal is starting
+			int startingColumn = -1;
+			int startingRow = -1; 
+			if(pieceColumn < column) {
+				startingColumn = pieceColumn;
+			}
+			if(pieceColumn > column) {
+				startingColumn = column;
+			}
+			if(pieceRow < row) {
+				startingRow = row;				
+			}
+			if(pieceRow > row) {
+				startingRow = row;				
+			}	
+			
+			//Implement through the squares the animal travels to check if the rat occupies the same space.
+			if(startingColumn < 0) {
+				while(startingColumn < startingColumn+3) {
+					if((gamePieces[0][1].getColumn() == startingColumn)	&& (gamePieces[0][1].getRow()==row)) {
+						return false;
+					}
+					if((gamePieces[1][1].getColumn() == startingColumn)	&& (gamePieces[1][1].getRow()==row)) {
+						return false;
+					}
+					startingColumn++;
+				}
+			} else {
+				while(startingRow < startingRow+4) {
+					if((gamePieces[0][1].getColumn() == column)	&& (gamePieces[0][1].getRow() == startingRow)) {
+						return false;
+					}
+					if((gamePieces[1][1].getColumn() == column)	&& (gamePieces[1][1].getRow( )== startingRow)) {
+						return false;
+					}
+					startingRow++;
+				}
+			}
+		}
+		
+		//At this point the piece is either moving 1 space or it's a tiger/leopard jumping a river
+		//check if it can land on the space. 
+		//if it's normal, trap, or den anything can be on it.
+		BoardSquare boardSquare = boardSquares[column][row];
+		BoardSquare pieceSquare = boardSquares[gamePiece.getColumn()][gamePiece.getRow()];
+		boolean enemyBanned = false;
+		if(boardSquare.getSquareType().equals(SquareType.River)) {
```

Comment:
You could handle this with polymorphism with something like `piece.canOccupy(BoardSquare)`

Reply:
This will again increase coupling while using the same code in a different spot.